pkg/operator/installer_controller_v311_00.go

```diff
@@ -0,0 +1,183 @@
+		if currKubeletState.CurrentDeploymentID != currKubeletState.TargetDeploymentID {
+			// TODO check to see if our installer pod completed.  Success or failure there indicates whether we should be failed.
+			newCurrKubeletState, err := checkIfInstallComplete(c.kubeClient, currKubeletState)
+			if err != nil {
+				return true, err
+			}
+
+			// if we make a change to this status, we want to write it out to the API before we commence work on the next kubelet.
+			// it's an extra write/read, but it makes the state debuggable from outside this process
+			if !equality.Semantic.DeepEqual(newCurrKubeletState, currKubeletState) {
+				glog.Infof("%q moving to %v", currKubeletState.NodeName, spew.Sdump(*newCurrKubeletState))
+				operatorConfig.Status.TargetKubeletStates[i] = *newCurrKubeletState
+				if !reflect.DeepEqual(operatorConfigOriginal, operatorConfig) {
+					_, updateError := c.operatorConfigClient.KubeApiserverOperatorConfigs().UpdateStatus(operatorConfig)
+					return false, updateError
+				}
+			}
+			break
+		}
+
+		deploymentIDToStart := getDeploymentIDToStart(currKubeletState, prevKubeletState, operatorConfig)
+		if deploymentIDToStart == 0 {
+			glog.V(4).Infof("%q does not need update", currKubeletState.NodeName)
+			continue
+		}
+		glog.Infof("%q needs to deploy to %d", currKubeletState.NodeName, deploymentIDToStart)
+
+		// we need to start a deployment create a pod that will lay down the static pod resources
+		newCurrKubeletState, err := createInstallerPod(c.kubeClient, currKubeletState, operatorConfig, deploymentIDToStart)
+		if err != nil {
+			return true, err
+		}
+		// if we make a change to this status, we want to write it out to the API before we commence work on the next kubelet.
+		// it's an extra write/read, but it makes the state debuggable from outside this process
+		if !equality.Semantic.DeepEqual(newCurrKubeletState, currKubeletState) {
+			glog.Infof("%q moving to %v", currKubeletState.NodeName, spew.Sdump(*newCurrKubeletState))
+			operatorConfig.Status.TargetKubeletStates[i] = *newCurrKubeletState
+			if !reflect.DeepEqual(operatorConfigOriginal, operatorConfig) {
+				_, updateError := c.operatorConfigClient.KubeApiserverOperatorConfigs().UpdateStatus(operatorConfig)
+				return false, updateError
+			}
+		}
+		break
+	}
+
+	v1alpha1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1alpha1.OperatorCondition{
+		Type:   "InstallerControllerFailing",
+		Status: operatorv1alpha1.ConditionFalse,
+	})
+	if !reflect.DeepEqual(operatorConfigOriginal, operatorConfig) {
+		_, updateError := c.operatorConfigClient.KubeApiserverOperatorConfigs().UpdateStatus(operatorConfig)
+		if updateError != nil {
+			return true, updateError
+		}
+	}
+
+	return false, nil
+}
+
+// checkIfInstallComplete returns the new KubeletState or and error
+func checkIfInstallComplete(c kubernetes.Interface, currKubeletState *v1alpha1.KubeletState) (*v1alpha1.KubeletState, error) {
```

Comment:
name is strange. `checkIf` suggests a boolean return value. Why not just `kubeletState(...)`?